cmd/thanos/sidecar.go

```diff
@@ -225,33 +227,36 @@ func runSidecar(
 		})
 	}
 
-	var uploads = true
-
 	bucketConfig, err := objStoreConfig.Content()
 	if err != nil {
 		return err
 	}
 
-	// The background shipper continuously scans the data directory and uploads
-	// new blocks to Google Cloud Storage or an S3-compatible storage service.
-	bkt, err := client.NewBucket(logger, bucketConfig, reg, component)
-	if err != nil && err != client.ErrNotFound {
-		return err
-	}
-
-	if err == client.ErrNotFound {
+	var uploads = true
+	if len(bucketConfig) == 0 {
 		level.Info(logger).Log("msg", "No supported bucket was configured, uploads will be disabled")
 		uploads = false
 	}
 
 	if uploads {
+		// The background shipper continuously scans the data directory and uploads
+		// new blocks to Google Cloud Storage or an S3-compatible storage service.
+		bkt, err := client.NewBucket(logger, bucketConfig, reg, component)
```

Comment:
pkg/objstore/client/factory.go has the BucketConfig struct while the bucketConfig here is a  byte[] that we hope holds some yaml data that might match the BucketConfig struct. Consider renaming to  `bucketConfig` -> `confContentYaml`to match NewBucket implementation.